tests/unit/s2n_rsa_pss_rsae_test.c

```diff
@@ -292,6 +292,8 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(BN_hex2bn(&d, test_case.key_param_d));
         EXPECT_SUCCESS(RSA_set0_key(rsa_key, n, e, d));
 
+        EVP_PKEY_set1_RSA(rsa_public_key.pkey, rsa_key);
```

Comment:
Why are we able to set `RSA *rsa_key` to what should be a `const RSA *rsa_key` return value, and is it okay to then pass that "cached copy" back with set1_RSA? We don't need to use get1_RSA here?

Reply:
Yes I think we do, because `get1` returns a non-const, but also a copy of the cached key. The compiler doesnt seem to apply Werror to the test files.. so I was able to get away with not declaring it as `const`.

Both `get0` and `get1` operate on the cached value so this was working fine.

It should be noted that the docs do state that values returned from these functions should be tread as constant since they operate on the cached value, however since usage only lasts till end of each loop we should be ok to overwrite the values.